### PR TITLE
Stan MCMC: prerequisites (1) -- ports from NOvA and nits

### PR DESCRIPTION
**Pull request by chenel**
This is the first in a series of pull requests introducing Bayesian MCMC using the Stan package (stan-mc.org).  (For the general goals and context, see the final PR, #7.)

This PR introduces a few prerequisites:
  (1) a few tools ported from NOvA CAFAna needed for the MCMC implementation, and the subsequent knock-on reorganizations
  (2) fixing a couple of nits to base CAFAna which made the Stan MCMC development easier.

### CODE REVIEW

File: CAFAna/Analysis/Plots.cxx [hunk 1]
```diff
@@ -30,7 +31,7 @@
 namespace ana
 {
   //----------------------------------------------------------------------
-  TH1* DataMCComparison(const Spectrum& data, const Spectrum& mc)
```
Comment (chenel) [inline CAFAna/Analysis/Plots.cxx]: Here's where `DataMCComparison()` needs to use the `EBinType` argument

File: CAFAna/Core/Progress.cxx [hunk 2]
```diff
@@ -21,7 +21,7 @@ namespace ana
     // If no else is drawing, we can
```
Comment (cjbackhouse) [inline CAFAna/Core/Progress.cxx]: This ProgressBar patch is the one we need on the nova side.

Comment (chenel) [reply CAFAna/Core/Progress.cxx]: thanks.  turns out I had this staged, and had just forgot to push it.  done.

File: CAFAna/Analysis/Plots.cxx (hunk 1 above)
Comment (chenel) [reply CAFAna/Analysis/Plots.cxx]: resolved with fe5b6aa4 below.

**Pull request merged**